aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -341,17 +341,19 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return result;
 }
 
-static inline void mvlgamma_check(const Tensor& self, int64_t p) {
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()),
-              "mvlgamma is not implemented for ", self.scalar_type());
+static inline void mvlgamma_check(const Tensor& self, int64_t p, bool floating_ufunc=false) {
+  if(!floating_ufunc)
+    TORCH_CHECK(at::isFloatingType(self.scalar_type()),
+            "mvlgamma is not implemented for ", self.scalar_type());
   TORCH_CHECK((self > 0.5f * (p - 1)).all().item<bool>(),
               "All elements must be greater than (p-1)/2");
   TORCH_CHECK(p >= 1, "p has to be greater than or equal to 1");
 }
 
 Tensor mvlgamma(const Tensor& self, int64_t p) {
-  mvlgamma_check(self, p);
-  Tensor args = native::arange(-p / 2. + 0.5, 0.5, 0.5, self.options());
+  mvlgamma_check(self, p, /*floating_ufunc=*/true);
```

Comment:
`true` is passed to ensure `mvlgamma_check` does not go through the `TORCH_CHECK(at::isFloatingType(self.scalar_type()), "mvlgamma is not implemented for ", self.scalar_type())` check.

Reply:
See above note: should we change mvlgamma at all?